frame/contracts/src/wasm/prepare.rs

```diff
@@ -313,22 +332,6 @@ impl<'a, T: Config> ContractModule<'a, T> {
 					imported_mem_type = Some(memory_type);
 					continue
 				},
-			};
-
-			let Type::Function(ref func_ty) = types
-				.get(*type_idx as usize)
-				.ok_or("validation: import entry points to a non-existent type")?;
-
-			if !T::ChainExtension::enabled() &&
-				import.field().as_bytes() == b"seal_call_chain_extension"
-			{
-				return Err("module uses chain extensions but chain extensions are disabled")
-			}
-
-			if import_fn_banlist.iter().any(|f| import.field().as_bytes() == *f) ||
-				!C::can_satisfy(import.module().as_bytes(), import.field().as_bytes(), func_ty)
-			{
-				return Err("module imports a non-existent function")
```

Comment:
So this well-known for contract authors error message will be replaced with the more unambiguous `new code rejected after instrumentation`, right?

I think we need to highlight this change better somewhere in docs + release notes, as it could make some frustration to the contract developers.

This is a quite often case when a contract using some new host function being deployed to an elder node, not supporting it, yet